docs/source/howto/exploring.rst

```diff
@@ -0,0 +1,137 @@
+
+To discover the neighbors of a given node, you can use the methods :meth:`~aiida.orm.nodes.node.Node.get_incoming` and :meth:`~aiida.orm.nodes.node.Node.get_outgoing`.
+They have the exact same interface but will return the neighbors from which the links are incoming, or to which the links are pointing, respectively.
+For example, for a given ``node``, to inspect all the neighboring nodes from which a link is incoming to the ``node``:
+
+.. code-block:: python
+
+    node.get_incoming()
+
+This will return an instance of the :class:`~aiida.orm.utils.links.LinkManager`.
+From that manager you can request the results in a specific format.
+If you are only interested in the neighboring nodes themselves, you can call the :class:`~aiida.orm.utils.links.LinkManager.all_nodes` method:
+
+.. code-block:: python
+
+    node.get_incoming().all_nodes()
+
+This will return a list of :class:`~aiida.orm.nodes.node.Node` instances that correspond to the nodes that are neighbors of ``node``, where the link is going towards ``node``.
+Calling the :meth:`~aiida.orm.utils.links.LinkManager.all` method of the manager instead will return a list of :class:`~aiida.orm.utils.links.LinkTriple` named tuples.
+These tuples contain, in addition to the neighboring node, also the link label and the link type with which they are connected to the origin ``node``.
+For example, to list all the neighbors of a node from which a link is incoming:
+
+.. code-block:: python
+
+    for link_triple in node.get_incoming().all():
+        print(link_triple.node, link_triple.link_type, link_triple.link_label)
+
+Note that the :class:`~aiida.orm.utils.links.LinkManager` provides many convenience methods to get information from the neigboring nodes, such as :meth:`~aiida.orm.utils.links.LinkManager.all_link_labels` if you only need the list of link labels.
+
+The :meth:`~aiida.orm.nodes.node.Node.get_incoming` and :meth:`~aiida.orm.nodes.node.Node.get_outgoing` methods accept various arguments that allow one to filter what neighboring nodes should be matched:
+
+ * ``node_class``: accepts a subclass of :class:`~aiida.orm.nodes.node.Node`, only neighboring nodes with a class that matches this will be returned
+ * ``link_type``: accepts a vale of :class:`~aiida.common.links.LinkType`, only neighboring nodes that are linked with this link type will be returned
+ * ``link_label_filter``: accepts a string regex (with optional wildcards), only neighboring nodes that are linked with a link label that matches the regex will be returned
+
+As an example:
+
+.. code-block:: python
+
+    node.get_incoming(node_class=Data, link_type=LinkType.INPUT_CALC, link_label_filter='output%node_').all_nodes()
+
+will return only neighboring data nodes that are linked to the ``node`` with a link of type ``LinkType.INPUT_CALC`` and where the link label matches the regex ``'output%node_'``.
+The ``%`` character matches one or more characters and the ``_`` character matches exactly one character.
```

Comment:
```suggestion
Reminder on the syntax of SQL `LIKE` patterns: The ``%`` character matches any string of zero or more characters, while the ``_`` character matches exactly one character. These two special characters can be escaped by prepending them with a backslash (note that when putting a backslash in a python string you have to escape the backslash itself, so you'll need two backslashes: e.g., to match exactly a link label `a_b` you need to pass `link_label_filter='a\\b'`).
```

Reply:
Did you mean `link_label_filter='a\\_b'` there in the last sentence? I think you may have forgotten the underscore